### PR TITLE
enhancement(host_metrics source): 'iowait' stat on Linux

### PR DESCRIPTION
**Pull request by charmitro**
Collect the `iowait` stat on Linux hosts using `heim`'s extension trait.

Closes: #13366

### CODE REVIEW

File: src/sources/host_metrics/cpu.rs [hunk 1]
```diff
@@ -34,18 +34,19 @@ impl HostMetrics {
                                 self.counter(
                                     name,
                                     timestamp,
-                                    times.nice().get::<second>(),
```
Comment (jszwedko) [inline src/sources/host_metrics/cpu.rs:37]: Did you mean to drop `nice` in this PR? I think the intent of the issue was just to _add_ iowait but leave the others.

Comment (charmitro) [reply src/sources/host_metrics/cpu.rs]: You're right, reverted.

File: src/sources/host_metrics/cpu.rs [hunk 2]
```diff
@@ -40,6 +40,16 @@ impl HostMetrics {
                                         (String::from("cpu"), index.to_string()),
                                     ]),
                                 ),
+                                #[cfg(target_os = "linux")]
+                                self.counter(
+                                    name,
+                                    timestamp,
+                                    times.io_wait().get::<second>(),
+                                    BTreeMap::from([
+                                        (String::from("mode"), String::from("io_wait")),
```
Comment (bruceg) [inline src/sources/host_metrics/cpu.rs:49]: This new mode should get added to the documentation in `website/cue/reference/components/sources/host_metrics.cue`.

**Review by bruceg — CHANGES_REQUESTED** [review]
Thanks, @charmitro. This looks good but is missing the relevant addition to the docs to cover this new mode.

File: src/sources/host_metrics/cpu.rs (hunk 2 above)
Comment (charmitro) [reply src/sources/host_metrics/cpu.rs:49]: Thanks for specifying this. Couldn't figure out the docs. 

Please let me know if there is anything else regarding this addition.

File: website/cue/reference/components/sources/host_metrics.cue [hunk 3]
```diff
@@ -430,7 +430,7 @@ components: sources: host_metrics: {
 				mode: {
 					description: "Which mode the CPU was running in during the given time."
 					required:    true
-					examples: ["idle", "system", "user", "nice"]
+					examples: ["idle", "system", "user", "nice", "io-wait"]
```
Comment (bruceg) [inline website/cue/reference/components/sources/host_metrics.cue:433]: Is this a typo? The code says `io_wait` here (underscore vs hyphen).

Comment (charmitro) [reply website/cue/reference/components/sources/host_metrics.cue]: Fixed typo, thanks for noticing it!

**Review by bruceg — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
Awesome, thanks for this contribution @charmitro !

**Pull request merged**